### PR TITLE
SmartR

### PR DESCRIPTION
**Pull request by PiotrZakrzewski**

### CODE REVIEW

File: src/app/SmartR/components/services/rServeService.js [hunk 1]
```diff
@@ -182,7 +182,9 @@ angular.module('smartRApp').factory('rServeService', [
                     taskType: taskData.taskType,
                     workflow: workflow
                 }
-            });
+            };
+            console.log(_httpArg);
```
Comment (berndvdveen) [inline src/app/SmartR/components/services/rServeService.js:229]: Please remove

File: src/app/SmartR/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        //disable IE ajax request caching //Causes error when used in combination with OAuth
+        //$httpProvider.defaults.headers.get['If-Modified-Since'] = 'Mon, 26 Jul 1997 05:00:00 GMT';
+
+        $locationProvider.html5Mode({
+            enabled: true,
+            requireBase: false
+        });
+
+    }]);
+    /*.run(function($rootScope, $http, masterEndpointConfig) {
```
Comment (kretep) [inline src/app/SmartR/index.js:20]: maybe remove?

File: src/app/SmartR/components/services/rServeService.js (hunk 1 above)
Comment (kretep) [reply src/app/SmartR/components/services/rServeService.js:229]: There seem to be more console.logs. Maybe remove them also.

File: src/app/SmartR/components/controllers/correlation.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            disabled: true,
+            running: false,
+            scriptResults: {},
+            params: {
+                method: 'pearson',
+                transformation: 'raw'
+            }
+        };
+
+        $scope.$watch('runAnalysis.params.transformation', function(newValue, oldValue) {
```
Comment (beaugogh) [inline src/app/SmartR/components/controllers/correlation.js:30]: why do we do a watch over transformation and switch correlation analysis? wouldn't watching a flag indicating the type of correlation (e.g. var corrType = 'pearson'/'spearman') be clearer?

File: src/app/SmartR/components/directives/biomarkerSelection.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            biomarkers: '='
+        },
+        templateUrl:  'app/SmartR/containers/templates/biomarkerSelection.html',
+        controller: function ($scope) {
+            if (!$scope.biomarkers) {
+                $scope.biomarkers = [];
+            }
+
+            var input = $('#heim-input-txt-identifier');
+            input.autocomplete({
```
Comment (beaugogh) [inline src/app/SmartR/components/directives/biomarkerSelection.js:20]: take out the functions from the nest, so that they are more readable?

**Pull request merged**